docs: add explanation for working multiple outputs to taskflow tutorial

cmarteepants opened the pull request:
Adds explanation for working with multiple outputs. 

Related: #45649

pierrejeambrun reviewed (commented):
If the note about `multiple_outputs` moving somewhere else ?

cmarteepants committed:
add callout for using multiple outputs in step 2 example

Accidentally omitted this when refactoring the tutorial.
It's now explicitly talked about in Step 2 since it's used in the example.
